### ISSUE TITLE
Swift Package Manager: project has unsupported layout

### ISSUE THREAD

**Issue by 4np** [SwiftyJSON/SwiftyJSON#709]
The project layout is incompatible with the Swift Package Manager.

### Steps to Reproduce

```
myUser ~ $ mkdir test
myUser ~ $ cd test
myUser ~/test $ printf "import PackageDescription\n\nlet package = Package(\n\tname: \"Test\",\n\ttargets: [],\n\tdependencies: [\n\t\t.Package(url: \"https://github.com/SwiftyJSON/SwiftyJSON.git\", majorVersion: 3)\n\t]\n)" > Package.swift
myUser ~/test $ swift package generate-xcodeproj
Cloning https://github.com/SwiftyJSON/SwiftyJSON.git
HEAD is now at 6a0c3a8 Merge pull request #681 from aterris/master
Resolved version: 3.1.1
error: the package has an unsupported layout, unexpected source file(s) found: /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/ArrayTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/BaseTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/ComparableTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/DictionaryTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/LiteralConvertibleTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/NestedJSONTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/NumberTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/PerformanceTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/PrintableTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/RawRepresentableTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/RawTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/SequenceTypeTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/StringTests.swift, /Users/myUser/test/Packages/SwiftyJSON-3.1.1/Tests/SubscriptTests.swift
fix: move the file(s) inside a module
myUser ~/test $ 
```

### Swift version

```
myUser ~/test $ swift --version
Apple Swift version 3.0.1 (swiftlang-800.0.58.6 clang-800.0.42.1)
Target: x86_64-apple-macosx10.9
myUser ~/test $ swift package --version
Apple Swift Package Manager - Swift 3.0.1 (swiftpm-11504)
myUser ~/test $ 
```

#705 suggests this should not a problem anymore... However I do not see a [version number bump](https://github.com/SwiftyJSON/SwiftyJSON/releases)? Or perhaps I just encountered this just after that issue was closed / pull request was merged... 🤔 In any case, I will leave this open until the version is fixed.

_related issues: #705, #688, #707?_

**Comment by rickw** [comment]
The layout error just started in Xcode 8.1 or Swift 3.0.1

**Comment by wongzigii** [comment]
For those who suffer this issue, you can use my branch as a workaround for now:

    .Package(url: "https://github.com/wongzigii/SwiftyJSON.git", majorVersion: 3, minor: 1)

**Comment by wongzigii** [comment]
@zhigang1992 Would you mind pushing a new release so that SPM could catch this up? 😃

**Comment by zhigang1992** [comment]
@wongzigii sure thing, just pushed `3.1.2` can you test it out and see if it works?

**Comment by wongzigii** [comment]
@zhigang1992 

    ➜  swift build
    Cloning https://github.com/SwiftyJSON/SwiftyJSON.git
    HEAD is now at 1dc2ce0 Merge pull request #699 from JeffGuKang/feature/Playground
    Resolved version: 3.1.2
    Compile Swift Module 'SwiftyJSON' (1 sources)
    Compile Swift Module 'MyProject' (1 sources)
    Linking ./.build/debug/MyProject

**Comment by wongzigii** [comment]
I believe this issue could be closed for now. :)

**Comment by 4np** [comment]
Confirmed fixed!

```
myUser $ swift package generate-xcodeproj
Cloning https://github.com/SwiftyJSON/SwiftyJSON.git
HEAD is now at 1dc2ce0 Merge pull request #699 from JeffGuKang/feature/Playground
Resolved version: 3.1.2
warning: root package 'Test' does not contain any sources
generated: ./Test.xcodeproj
myUser $
```
